pkg/codegen/convert/plugin_mapper.go

```diff
@@ -173,12 +173,87 @@ func NewPluginMapper(ws Workspace,
 	}, nil
 }
 
+// getMappingForPlugin calls GetMapping on the given plugin and returns it's result. This currently takes the
+// target provider name as an argument so that if we don't find the mapping for "terraform" we can try again
+// for "tf".
+func (l *pluginMapper) getMappingForPlugin(targetProvider string, pluginSpec mapperPluginSpec) ([]byte, string, error) {
+	providerPlugin, err := l.providerFactory(pluginSpec.name, &pluginSpec.version)
+	if err != nil {
+		// We should maybe be lenient here and ignore errors but for now assume it's better to fail out on
+		// things like providers failing to start.
+		return nil, "", fmt.Errorf("could not create provider '%s': %w", pluginSpec.name, err)
+	}
+	contract.IgnoreClose(providerPlugin)
+
+	data, mappedProvider, err := providerPlugin.GetMapping(l.conversionKey)
+	if err != nil {
+		// This was an error calling GetMapping, not just that GetMapping returned a nil result. It's fine
+		// for GetMapping to return (nil, nil).
```

Comment:
Can you adjust the comment to answer "why is it fine?"
I presume it's fine because `nil, nil` means "no mapping and no error", but it would be helpful to have that in the comment.